### ISSUE TITLE
Unable to delete files on Windows

### ISSUE THREAD

**Issue by smashwilson** [atom/atom#15507]
Attempting to delete files through Atom fails with a dialog:

<img width="394" alt="error" src="https://user-images.githubusercontent.com/17565/29887127-6d8545c6-8d8a-11e7-9d58-e67e7f4a62fa.png">

But they can be deleted normally from the command line:

```ps
C:\Users\smash\src\watcher [master ≡ +1 ~0 -0 !]
λ  remove-item .\src\worker\windows\foo.md
C:\Users\smash\src\watcher [master ≡]
λ
```

Or from Explorer, for that matter.

It seems to be happening only with files that are not yet tracked in a git repository.

/cc @ungb @iolsen @damieng

**Comment by smashwilson** [comment]
I can replicate this with the github package disabled.

Interestingly, in the dev console, `shell.moveItemToTrash()` fails, while `fs.unlink()` succeeds fine:

```js
> const {shell} = require('electron')
undefined
> const fs = require('fs')
undefined
> shell.moveItemToTrash('C:\\Users\\smash\\src\\atom\\src\\foo.md')
false
> fs.unlink('C:\\Users\\smash\\src\\atom\\src\\foo.md', () => {})
undefined
```

I'm not seeing anything obvious in [the Electron source for `moveItemToTrash`](https://github.com/electron/electron/blob/f0c38b7354fa0890950dd92550e8a8ff858a6dbb/atom/common/platform_util_win.cc#L325-L373), either.

**Comment by smashwilson** [comment]
I think I've figured it out.

Back in atom/node-spellchecker#75 and atom/spell-check#215, I introduced a COM initialization call to share the COM connection among many threads:

```cpp
CoInitializeEx(NULL, COINIT_MULTITHREADED)
``` 

From the [MSDN documentation for `CoInitializeEx()`](https://msdn.microsoft.com/en-us/library/windows/desktop/ms695279(v=vs.85).aspx):

> Multiple calls to CoInitializeEx by the same thread are allowed as long as they pass the same concurrency flag, but subsequent valid calls return S_FALSE.

> After the concurrency model for a thread is set, it cannot be changed. A call to CoInitialize on an apartment that was previously initialized as multithreaded will fail and return RPC_E_CHANGED_MODE.

If the spellchecker module is activated by, for example, opening a `.md` file, node-spellchecker is initializing COM in multithreaded mode. The COM connection managed by [`base::win::ScopedCOMInitializer` is created incompatibly](https://chromium.googlesource.com/chromium/+/07ca5b139b967f62b03f08fca5d6f37a373ed20f/base/win/scoped_com_initializer.h#28), so `com_initializer.succeeded()` is false.

I've tentatively verified this by running `apm disable spell-check` and observing that it no longer occurs, then running `apm enable spell-check`, opening a markdown file, and observing that it does occur.

I'll need to rethink the threading model used in node-spellchecker to find a way to interact with COM correctly, I think, or at least not hold the COM library open indefinitely.
